## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :x: [more...](#82976513-a24c-4b2a-a6c9-a546ace72384) |
| mint-fs.sh | :x: [more...](#93fa822d-f30d-4861-b3b7-6c9ec2f073a8) |
| mint-erasure.sh | :x: [more...](#8bae96c5-1639-4e07-98b3-680bc8efa091) |
| mint-dist-erasure.sh | :x: [more...](#313f0f58-4f5b-4c94-8477-9e3fa53529cc) |
| mint-compress-encrypt-dist-erasure.sh | :x: [more...](#29349102-14ba-496c-8200-cc8a31410678) |
| mint-pools.sh | :x: [more...](#645cb4fe-72a2-4c11-8d0e-8a65df224bfa) |
<a name="313f0f58-4f5b-4c94-8477-9e3fa53529cc"></a>
### 15685-2c4b8ef/mint-dist-erasure.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-k8s2:30564
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0
RUN_ON_FAIL:          0

To get logs, run 'docker cp :/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 1 seconds
(2/14) Running aws-sdk-java tests ... done in 1 seconds
(3/14) Running aws-sdk-php tests ... done in 44 seconds
(4/14) Running aws-sdk-ruby tests ... done in 5 seconds
(5/14) Running awscli tests ... done in 53 seconds
(6/14) Running healthcheck tests ... done in 0 seconds
(7/14) Running mc tests ... done in 26 seconds
(8/14) Running minio-go tests ... FAILED in 23 seconds
{
  "args": {
    "bucketName": "minio-go-test-z3soqj9h0hh36ddc",
    "objectName": "sy002cajrvirmy6pwpsd1qkd93ck1e"
  },
  "duration": 5464,
  "error": "We encountered an internal error, please try again.",
  "function": "GetObject()",
  "message": "PutObject failed",
  "name": "minio-go: testPutObjectWithVersioning",
  "status": "FAIL"
}
(8/14) Running minio-java tests ... FAILED in 2 seconds
{
  "name": "minio-java",
  "function": "setBucketVersioning()",
  "duration": 186,
  "status": "FAIL",
  "error": "error occurred\nErrorResponse(code = InvalidBucketState, message = A replication configuration is present on this bucket, bucket wide versioning cannot be suspended., bucketName = minio-java-test-2nh6dbe, objectName = null, resource = /minio-java-test-2nh6dbe, requestId = 171489A66715D82B, hostId = 7d425b42-7a7c-4ab8-8838-65a740fceeeb)\nrequest={method=PUT, url=http://minio-k8s2:30564/minio-java-test-2nh6dbe?versioning=, headers=Host: minio-k8s2:30564\nAccept-Encoding: identity\nUser-Agent: MinIO (Linux; amd64) minio-java/8.4.3\nContent-MD5: orZUUp7E9srl53Od8p1glA==\nx-amz-content-sha256: d7f1d7ed35d188f6a89d394cc2055f5910c14a3abccf41e808e7f7327d21e129\nx-amz-date: 20220913T214510Z\nAuthorization: AWS4-HMAC-SHA256 Credential=*REDACTED*/20220913/us-east-1/s3/aws4_request, SignedHeaders=content-md5;host;x-amz-content-sha256;x-amz-date, Signature=*REDACTED*\n}\nresponse={code=400, headers=Accept-Ranges: bytes\nContent-Length: 387\nContent-Security-Policy: block-all-mixed-content\nContent-Type: application/xml\nServer: MinIO\nStrict-Transport-Security: max-age=31536000; includeSubDomains\nVary: Origin\nVary: Accept-Encoding\nX-Amz-Request-Id: 171489A66715D82B\nX-Content-Type-Options: nosniff\nX-Xss-Protection: 1; mode=block\nDate: Tue, 13 Sep 2022 21:45:10 GMT\n}\n >>> [io.minio.S3Base$1.onResponse(S3Base.java:690), okhttp3.internal.connection.RealCall$AsyncCall.run(RealCall.kt:519), java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128), java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628), java.base/java.lang.Thread.run(Thread.java:829)]"
}
(8/14) Running minio-js tests ... FAILED in 51 seconds
{
  "name": "minio-js",
  "function": "Suspend versioning  on a bucket",
  "duration": 5,
  "status": "FAIL",
  "error": "S3Error: A replication configuration is present on this bucket, bucket wide versioning cannot be suspended. at Object.parseError (node_modules/minio/dist/main/xml-parsers.js:71:11) at /mint/run/core/minio-js/node_modules/minio/dist/main/transformers.js:166:22 at DestroyableTransform._flush (node_modules/minio/dist/main/transformers.js:90:10) at DestroyableTransform.prefinish (node_modules/readable-stream/lib/_stream_transform.js:129:10) at prefinish (node_modules/readable-stream/lib/_stream_writable.js:611:14) at finishMaybe (node_modules/readable-stream/lib/_stream_writable.js:620:5) at endWritable (node_modules/readable-stream/lib/_stream_writable.js:643:3) at DestroyableTransform.Writable.end (node_modules/readable-stream/lib/_stream_writable.js:571:22) at IncomingMessage.onend (internal/streams/readable.js:670:10) at endReadableNT (internal/streams/readable.js:1333:12) at processTicksAndRejections (internal/process/task_queues.js:82:21)"
}
(8/14) Running minio-py tests ... FAILED in 19 seconds
{
  "name": "minio-py:test_stat_object_version",
  "status": "FAIL",
  "args": {
    "bucket_name": "minio-py-test-93ace169-4882-4056-bf87-79b6f6f49603",
    "object_name": "6d63da9a-23bf-4855-81a6-2761de0d455e",
    "length": 1048576,
    "data": "LimitedRandomReader(1 * MB)"
  },
  "message": "HTTPConnectionPool(host='minio-k8s2', port=30564): Max retries exceeded with url: /minio-py-test-93ace169-4882-4056-bf87-79b6f6f49603/6d63da9a-23bf-4855-81a6-2761de0d455e (Caused by ResponseError('too many 500 error responses'))",
  "error": "Traceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 126, in _call_test\n    func(log_entry, *args, **kwargs)\n  File \"/mint/run/core/minio-py/tests.py\", line 825, in test_stat_object_version\n    _test_stat_object(log_entry, sse, version_check=True)\n  File \"/mint/run/core/minio-py/tests.py\", line 777, in _test_stat_object\n    result = _CLIENT.put_object(\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 1755, in put_object\n    raise exc\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 1709, in put_object\n    return self._put_object(\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 1567, in _put_object\n    response = self._execute(\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 403, in _execute\n    return self._url_open(\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 266, in _url_open\n    response = self._http.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/poolmanager.py\", line 376, in urlopen\n    response = conn.urlopen(method, u.request_uri, **kw)\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 878, in urlopen\n    return self.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 878, in urlopen\n    return self.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 878, in urlopen\n    return self.urlopen(\n  [Previous line repeated 2 more times]\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 868, in urlopen\n    retries = retries.increment(method, url, response=response, _pool=self)\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/util/retry.py\", line 592, in increment\n    raise MaxRetryError(_pool, url, error or ResponseError(cause))\nurllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='minio-k8s2', port=30564): Max retries exceeded with url: /minio-py-test-93ace169-4882-4056-bf87-79b6f6f49603/6d63da9a-23bf-4855-81a6-2761de0d455e (Caused by ResponseError('too many 500 error responses'))\n",
  "duration": 6546
}
(8/14) Running s3cmd tests ... done in 35 seconds
(9/14) Running s3select tests ... done in 4 seconds
(10/14) Running versioning tests ... FAILED in 0 seconds
{
  "alert": "",
  "args": {
    "bucketName": "versioning-test-j4dny9l164j2fq",
    "expiry": 60000000000,
    "objectName": "testObject"
  },
  "duration": 515,
  "error": "InternalError: We encountered an internal error, please try again.\n\tstatus code: 500, request id: 171489BFE6FE3A6B, host id: ",
  "function": "/mint/build/versioning/put.go:83: main.testPutObject",
  "message": "PUT expected to succeed but got InternalError: We encountered an internal error, please try again.\n\tstatus code: 500, request id: 171489BFE6FE3A6B, host id: ",
  "name": "versioning",
  "status": "FAIL"
}

Executed 9 out of 14 tests successfully.
```
<a name="645cb4fe-72a2-4c11-8d0e-8a65df224bfa"></a>
### 15685-2c4b8ef/mint-pools.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-k8s6:30440
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0
RUN_ON_FAIL:          0

To get logs, run 'docker cp :/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 7 seconds
(2/14) Running aws-sdk-java tests ... done in 1 seconds
(3/14) Running aws-sdk-php tests ... done in 43 seconds
(4/14) Running aws-sdk-ruby tests ... done in 5 seconds
(5/14) Running awscli tests ... done in 54 seconds
(6/14) Running healthcheck tests ... done in 0 seconds
(7/14) Running mc tests ... done in 27 seconds
(8/14) Running minio-go tests ... FAILED in 22 seconds
{
  "args": {
    "bucketName": "minio-go-test-mo4m2pjr1f6mcqvh",
    "objectName": "pjqr5q4z1jmzhtcq1lrzktz3o9kqq3"
  },
  "duration": 5567,
  "error": "We encountered an internal error, please try again.",
  "function": "GetObject()",
  "message": "PutObject failed",
  "name": "minio-go: testPutObjectWithVersioning",
  "status": "FAIL"
}
(8/14) Running minio-java tests ... FAILED in 2 seconds
{
  "name": "minio-java",
  "function": "setBucketVersioning()",
  "duration": 209,
  "status": "FAIL",
  "error": "error occurred\nErrorResponse(code = InvalidBucketState, message = A replication configuration is present on this bucket, bucket wide versioning cannot be suspended., bucketName = minio-java-test-pddvpq, objectName = null, resource = /minio-java-test-pddvpq, requestId = 171489A8479C72C2, hostId = 2395b001-8344-431a-a35c-55e7a93eedf6)\nrequest={method=PUT, url=http://minio-k8s6:30440/minio-java-test-pddvpq?versioning=, headers=Host: minio-k8s6:30440\nAccept-Encoding: identity\nUser-Agent: MinIO (Linux; amd64) minio-java/8.4.3\nContent-MD5: orZUUp7E9srl53Od8p1glA==\nx-amz-content-sha256: d7f1d7ed35d188f6a89d394cc2055f5910c14a3abccf41e808e7f7327d21e129\nx-amz-date: 20220913T214518Z\nAuthorization: AWS4-HMAC-SHA256 Credential=*REDACTED*/20220913/us-east-1/s3/aws4_request, SignedHeaders=content-md5;host;x-amz-content-sha256;x-amz-date, Signature=*REDACTED*\n}\nresponse={code=400, headers=Accept-Ranges: bytes\nContent-Length: 385\nContent-Security-Policy: block-all-mixed-content\nContent-Type: application/xml\nServer: MinIO\nStrict-Transport-Security: max-age=31536000; includeSubDomains\nVary: Origin\nVary: Accept-Encoding\nX-Amz-Request-Id: 171489A8479C72C2\nX-Content-Type-Options: nosniff\nX-Xss-Protection: 1; mode=block\nDate: Tue, 13 Sep 2022 21:45:18 GMT\n}\n >>> [io.minio.S3Base$1.onResponse(S3Base.java:690), okhttp3.internal.connection.RealCall$AsyncCall.run(RealCall.kt:519), java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128), java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628), java.base/java.lang.Thread.run(Thread.java:829)]"
}
(8/14) Running minio-js tests ... FAILED in 51 seconds
{
  "name": "minio-js",
  "function": "Suspend versioning  on a bucket",
  "duration": 4,
  "status": "FAIL",
  "error": "S3Error: A replication configuration is present on this bucket, bucket wide versioning cannot be suspended. at Object.parseError (node_modules/minio/dist/main/xml-parsers.js:71:11) at /mint/run/core/minio-js/node_modules/minio/dist/main/transformers.js:166:22 at DestroyableTransform._flush (node_modules/minio/dist/main/transformers.js:90:10) at DestroyableTransform.prefinish (node_modules/readable-stream/lib/_stream_transform.js:129:10) at prefinish (node_modules/readable-stream/lib/_stream_writable.js:611:14) at finishMaybe (node_modules/readable-stream/lib/_stream_writable.js:620:5) at endWritable (node_modules/readable-stream/lib/_stream_writable.js:643:3) at DestroyableTransform.Writable.end (node_modules/readable-stream/lib/_stream_writable.js:571:22) at IncomingMessage.onend (internal/streams/readable.js:670:10) at endReadableNT (internal/streams/readable.js:1333:12) at processTicksAndRejections (internal/process/task_queues.js:82:21)"
}
(8/14) Running minio-py tests ... FAILED in 20 seconds
{
  "name": "minio-py:test_stat_object_version",
  "status": "FAIL",
  "args": {
    "bucket_name": "minio-py-test-86428530-af63-40d5-af95-614a8ee0816d",
    "object_name": "2dc31d50-e83c-4272-b2f3-2e2546477b88",
    "length": 1048576,
    "data": "LimitedRandomReader(1 * MB)"
  },
  "message": "HTTPConnectionPool(host='minio-k8s6', port=30440): Max retries exceeded with url: /minio-py-test-86428530-af63-40d5-af95-614a8ee0816d/2dc31d50-e83c-4272-b2f3-2e2546477b88 (Caused by ResponseError('too many 500 error responses'))",
  "error": "Traceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 126, in _call_test\n    func(log_entry, *args, **kwargs)\n  File \"/mint/run/core/minio-py/tests.py\", line 825, in test_stat_object_version\n    _test_stat_object(log_entry, sse, version_check=True)\n  File \"/mint/run/core/minio-py/tests.py\", line 777, in _test_stat_object\n    result = _CLIENT.put_object(\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 1755, in put_object\n    raise exc\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 1709, in put_object\n    return self._put_object(\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 1567, in _put_object\n    response = self._execute(\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 403, in _execute\n    return self._url_open(\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 266, in _url_open\n    response = self._http.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/poolmanager.py\", line 376, in urlopen\n    response = conn.urlopen(method, u.request_uri, **kw)\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 878, in urlopen\n    return self.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 878, in urlopen\n    return self.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 878, in urlopen\n    return self.urlopen(\n  [Previous line repeated 2 more times]\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 868, in urlopen\n    retries = retries.increment(method, url, response=response, _pool=self)\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/util/retry.py\", line 592, in increment\n    raise MaxRetryError(_pool, url, error or ResponseError(cause))\nurllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='minio-k8s6', port=30440): Max retries exceeded with url: /minio-py-test-86428530-af63-40d5-af95-614a8ee0816d/2dc31d50-e83c-4272-b2f3-2e2546477b88 (Caused by ResponseError('too many 500 error responses'))\n",
  "duration": 6558
}
(8/14) Running s3cmd tests ... done in 25 seconds
(9/14) Running s3select tests ... done in 4 seconds
(10/14) Running versioning tests ... FAILED in 1 seconds
{
  "alert": "",
  "args": {
    "bucketName": "versioning-test-43b06a6cegc1r6",
    "expiry": 60000000000,
    "objectName": "testObject"
  },
  "duration": 390,
  "error": "InternalError: We encountered an internal error, please try again.\n\tstatus code: 500, request id: 171489BFDD2E6A3D, host id: ",
  "function": "/mint/build/versioning/put.go:83: main.testPutObject",
  "message": "PUT expected to succeed but got InternalError: We encountered an internal error, please try again.\n\tstatus code: 500, request id: 171489BFDD2E6A3D, host id: ",
  "name": "versioning",
  "status": "FAIL"
}

Executed 9 out of 14 tests successfully.
```
<a name="93fa822d-f30d-4861-b3b7-6c9ec2f073a8"></a>
### 15685-2c4b8ef/mint-fs.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-k8s3:32675
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0
RUN_ON_FAIL:          0

To get logs, run 'docker cp :/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 8 seconds
(2/14) Running aws-sdk-java tests ... done in 1 seconds
(3/14) Running aws-sdk-php tests ... done in 49 seconds
(4/14) Running aws-sdk-ruby tests ... done in 4 seconds
(5/14) Running awscli tests ... done in 50 seconds
(6/14) Running healthcheck tests ... done in 0 seconds
(7/14) Running mc tests ... done in 22 seconds
(8/14) Running minio-go tests ... FAILED in 21 seconds
{
  "args": {
    "bucketName": "minio-go-test-ipnk4r3hpy5zjyrv",
    "objectName": "iyk9pvr0o0xst040g2ds4n9zkyf9z6"
  },
  "duration": 4488,
  "error": "We encountered an internal error, please try again.",
  "function": "GetObject()",
  "message": "PutObject failed",
  "name": "minio-go: testPutObjectWithVersioning",
  "status": "FAIL"
}
(8/14) Running minio-java tests ... FAILED in 1 seconds
{
  "name": "minio-java",
  "function": "setBucketVersioning()",
  "duration": 62,
  "status": "FAIL",
  "error": "error occurred\nErrorResponse(code = InvalidBucketState, message = A replication configuration is present on this bucket, bucket wide versioning cannot be suspended., bucketName = minio-java-test-gcpnqa, objectName = null, resource = /minio-java-test-gcpnqa, requestId = 171489A3BC004187, hostId = 1771761d-1ddf-4c18-a948-9e272c7b6234)\nrequest={method=PUT, url=http://minio-k8s3:32675/minio-java-test-gcpnqa?versioning=, headers=Host: minio-k8s3:32675\nAccept-Encoding: identity\nUser-Agent: MinIO (Linux; amd64) minio-java/8.4.3\nContent-MD5: orZUUp7E9srl53Od8p1glA==\nx-amz-content-sha256: d7f1d7ed35d188f6a89d394cc2055f5910c14a3abccf41e808e7f7327d21e129\nx-amz-date: 20220913T214458Z\nAuthorization: AWS4-HMAC-SHA256 Credential=*REDACTED*/20220913/us-east-1/s3/aws4_request, SignedHeaders=content-md5;host;x-amz-content-sha256;x-amz-date, Signature=*REDACTED*\n}\nresponse={code=400, headers=Accept-Ranges: bytes\nContent-Length: 385\nContent-Security-Policy: block-all-mixed-content\nContent-Type: application/xml\nServer: MinIO\nStrict-Transport-Security: max-age=31536000; includeSubDomains\nVary: Origin\nVary: Accept-Encoding\nX-Amz-Request-Id: 171489A3BC004187\nX-Content-Type-Options: nosniff\nX-Xss-Protection: 1; mode=block\nDate: Tue, 13 Sep 2022 21:44:58 GMT\n}\n >>> [io.minio.S3Base$1.onResponse(S3Base.java:690), okhttp3.internal.connection.RealCall$AsyncCall.run(RealCall.kt:519), java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128), java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628), java.base/java.lang.Thread.run(Thread.java:829)]"
}
(8/14) Running minio-js tests ... FAILED in 49 seconds
{
  "name": "minio-js",
  "function": "Suspend versioning  on a bucket",
  "duration": 4,
  "status": "FAIL",
  "error": "S3Error: A replication configuration is present on this bucket, bucket wide versioning cannot be suspended. at Object.parseError (node_modules/minio/dist/main/xml-parsers.js:71:11) at /mint/run/core/minio-js/node_modules/minio/dist/main/transformers.js:166:22 at DestroyableTransform._flush (node_modules/minio/dist/main/transformers.js:90:10) at DestroyableTransform.prefinish (node_modules/readable-stream/lib/_stream_transform.js:129:10) at prefinish (node_modules/readable-stream/lib/_stream_writable.js:611:14) at finishMaybe (node_modules/readable-stream/lib/_stream_writable.js:620:5) at endWritable (node_modules/readable-stream/lib/_stream_writable.js:643:3) at DestroyableTransform.Writable.end (node_modules/readable-stream/lib/_stream_writable.js:571:22) at IncomingMessage.onend (internal/streams/readable.js:670:10) at endReadableNT (internal/streams/readable.js:1333:12) at processTicksAndRejections (internal/process/task_queues.js:82:21)"
}
(8/14) Running minio-py tests ... FAILED in 18 seconds
{
  "name": "minio-py:test_stat_object_version",
  "status": "FAIL",
  "args": {
    "bucket_name": "minio-py-test-1238fa16-b7ff-471f-9793-ba0db3f6f755",
    "object_name": "c7a11177-1443-4415-8375-04d4fdffa38d",
    "length": 1048576,
    "data": "LimitedRandomReader(1 * MB)"
  },
  "message": "HTTPConnectionPool(host='minio-k8s3', port=32675): Max retries exceeded with url: /minio-py-test-1238fa16-b7ff-471f-9793-ba0db3f6f755/c7a11177-1443-4415-8375-04d4fdffa38d (Caused by ResponseError('too many 500 error responses'))",
  "error": "Traceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 126, in _call_test\n    func(log_entry, *args, **kwargs)\n  File \"/mint/run/core/minio-py/tests.py\", line 825, in test_stat_object_version\n    _test_stat_object(log_entry, sse, version_check=True)\n  File \"/mint/run/core/minio-py/tests.py\", line 777, in _test_stat_object\n    result = _CLIENT.put_object(\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 1755, in put_object\n    raise exc\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 1709, in put_object\n    return self._put_object(\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 1567, in _put_object\n    response = self._execute(\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 403, in _execute\n    return self._url_open(\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 266, in _url_open\n    response = self._http.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/poolmanager.py\", line 376, in urlopen\n    response = conn.urlopen(method, u.request_uri, **kw)\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 878, in urlopen\n    return self.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 878, in urlopen\n    return self.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 878, in urlopen\n    return self.urlopen(\n  [Previous line repeated 2 more times]\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 868, in urlopen\n    retries = retries.increment(method, url, response=response, _pool=self)\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/util/retry.py\", line 592, in increment\n    raise MaxRetryError(_pool, url, error or ResponseError(cause))\nurllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='minio-k8s3', port=32675): Max retries exceeded with url: /minio-py-test-1238fa16-b7ff-471f-9793-ba0db3f6f755/c7a11177-1443-4415-8375-04d4fdffa38d (Caused by ResponseError('too many 500 error responses'))\n",
  "duration": 6410
}
(8/14) Running s3cmd tests ... done in 23 seconds
(9/14) Running s3select tests ... done in 1 seconds
(10/14) Running versioning tests ... FAILED in 1 seconds
{
  "alert": "",
  "args": {
    "bucketName": "versioning-test-uosb3idrt62d1u",
    "expiry": 60000000000,
    "objectName": "testObject"
  },
  "duration": 337,
  "error": "InternalError: We encountered an internal error, please try again.\n\tstatus code: 500, request id: 171489B9029A51D1, host id: ",
  "function": "/mint/build/versioning/put.go:83: main.testPutObject",
  "message": "PUT expected to succeed but got InternalError: We encountered an internal error, please try again.\n\tstatus code: 500, request id: 171489B9029A51D1, host id: ",
  "name": "versioning",
  "status": "FAIL"
}

Executed 9 out of 14 tests successfully.
```
<a name="8bae96c5-1639-4e07-98b3-680bc8efa091"></a>
### 15685-2c4b8ef/mint-erasure.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-k8s2:30332
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0
RUN_ON_FAIL:          0

To get logs, run 'docker cp :/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 8 seconds
(2/14) Running aws-sdk-java tests ... done in 1 seconds
(3/14) Running aws-sdk-php tests ... done in 1 minutes and 7 seconds
(4/14) Running aws-sdk-ruby tests ... done in 4 seconds
(5/14) Running awscli tests ... done in 50 seconds
(6/14) Running healthcheck tests ... done in 0 seconds
(7/14) Running mc tests ... done in 21 seconds
(8/14) Running minio-go tests ... FAILED in 19 seconds
{
  "args": {
    "bucketName": "minio-go-test-s4pmt6wwbnpmqy9j",
    "objectName": "f35gpkw30655vioxeyekgfsphyumjm"
  },
  "duration": 5194,
  "error": "We encountered an internal error, please try again.",
  "function": "GetObject()",
  "message": "PutObject failed",
  "name": "minio-go: testPutObjectWithVersioning",
  "status": "FAIL"
}
(8/14) Running minio-java tests ... FAILED in 2 seconds
{
  "name": "minio-java",
  "function": "setBucketVersioning()",
  "duration": 63,
  "status": "FAIL",
  "error": "error occurred\nErrorResponse(code = InvalidBucketState, message = A replication configuration is present on this bucket, bucket wide versioning cannot be suspended., bucketName = minio-java-test-15q9h47, objectName = null, resource = /minio-java-test-15q9h47, requestId = 171489A26B9292F3, hostId = 98ab7ff9-9573-4437-9e12-513d0ae30205)\nrequest={method=PUT, url=http://minio-k8s2:30332/minio-java-test-15q9h47?versioning=, headers=Host: minio-k8s2:30332\nAccept-Encoding: identity\nUser-Agent: MinIO (Linux; amd64) minio-java/8.4.3\nContent-MD5: orZUUp7E9srl53Od8p1glA==\nx-amz-content-sha256: d7f1d7ed35d188f6a89d394cc2055f5910c14a3abccf41e808e7f7327d21e129\nx-amz-date: 20220913T214453Z\nAuthorization: AWS4-HMAC-SHA256 Credential=*REDACTED*/20220913/us-east-1/s3/aws4_request, SignedHeaders=content-md5;host;x-amz-content-sha256;x-amz-date, Signature=*REDACTED*\n}\nresponse={code=400, headers=Accept-Ranges: bytes\nContent-Length: 387\nContent-Security-Policy: block-all-mixed-content\nContent-Type: application/xml\nServer: MinIO\nStrict-Transport-Security: max-age=31536000; includeSubDomains\nVary: Origin\nVary: Accept-Encoding\nX-Amz-Request-Id: 171489A26B9292F3\nX-Content-Type-Options: nosniff\nX-Xss-Protection: 1; mode=block\nDate: Tue, 13 Sep 2022 21:44:53 GMT\n}\n >>> [io.minio.S3Base$1.onResponse(S3Base.java:690), okhttp3.internal.connection.RealCall$AsyncCall.run(RealCall.kt:519), java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128), java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628), java.base/java.lang.Thread.run(Thread.java:829)]"
}
(8/14) Running minio-js tests ... FAILED in 49 seconds
{
  "name": "minio-js",
  "function": "Suspend versioning  on a bucket",
  "duration": 5,
  "status": "FAIL",
  "error": "S3Error: A replication configuration is present on this bucket, bucket wide versioning cannot be suspended. at Object.parseError (node_modules/minio/dist/main/xml-parsers.js:71:11) at /mint/run/core/minio-js/node_modules/minio/dist/main/transformers.js:166:22 at DestroyableTransform._flush (node_modules/minio/dist/main/transformers.js:90:10) at DestroyableTransform.prefinish (node_modules/readable-stream/lib/_stream_transform.js:129:10) at prefinish (node_modules/readable-stream/lib/_stream_writable.js:611:14) at finishMaybe (node_modules/readable-stream/lib/_stream_writable.js:620:5) at endWritable (node_modules/readable-stream/lib/_stream_writable.js:643:3) at DestroyableTransform.Writable.end (node_modules/readable-stream/lib/_stream_writable.js:571:22) at IncomingMessage.onend (internal/streams/readable.js:670:10) at endReadableNT (internal/streams/readable.js:1333:12) at processTicksAndRejections (internal/process/task_queues.js:82:21)"
}
(8/14) Running minio-py tests ... FAILED in 18 seconds
{
  "name": "minio-py:test_stat_object_version",
  "status": "FAIL",
  "args": {
    "bucket_name": "minio-py-test-cd415fcf-c4f3-4551-a9ff-d7a1bc0dd4a4",
    "object_name": "d995bd63-8567-400f-826e-b4dfb2193cd9",
    "length": 1048576,
    "data": "LimitedRandomReader(1 * MB)"
  },
  "message": "HTTPConnectionPool(host='minio-k8s2', port=30332): Max retries exceeded with url: /minio-py-test-cd415fcf-c4f3-4551-a9ff-d7a1bc0dd4a4/d995bd63-8567-400f-826e-b4dfb2193cd9 (Caused by ResponseError('too many 500 error responses'))",
  "error": "Traceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 126, in _call_test\n    func(log_entry, *args, **kwargs)\n  File \"/mint/run/core/minio-py/tests.py\", line 825, in test_stat_object_version\n    _test_stat_object(log_entry, sse, version_check=True)\n  File \"/mint/run/core/minio-py/tests.py\", line 777, in _test_stat_object\n    result = _CLIENT.put_object(\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 1755, in put_object\n    raise exc\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 1709, in put_object\n    return self._put_object(\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 1567, in _put_object\n    response = self._execute(\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 403, in _execute\n    return self._url_open(\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 266, in _url_open\n    response = self._http.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/poolmanager.py\", line 376, in urlopen\n    response = conn.urlopen(method, u.request_uri, **kw)\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 878, in urlopen\n    return self.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 878, in urlopen\n    return self.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 878, in urlopen\n    return self.urlopen(\n  [Previous line repeated 2 more times]\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 868, in urlopen\n    retries = retries.increment(method, url, response=response, _pool=self)\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/util/retry.py\", line 592, in increment\n    raise MaxRetryError(_pool, url, error or ResponseError(cause))\nurllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='minio-k8s2', port=30332): Max retries exceeded with url: /minio-py-test-cd415fcf-c4f3-4551-a9ff-d7a1bc0dd4a4/d995bd63-8567-400f-826e-b4dfb2193cd9 (Caused by ResponseError('too many 500 error responses'))\n",
  "duration": 6423
}
(8/14) Running s3cmd tests ... done in 28 seconds
(9/14) Running s3select tests ... done in 2 seconds
(10/14) Running versioning tests ... FAILED in 1 seconds
{
  "alert": "",
  "args": {
    "bucketName": "versioning-test-5jit2sjhqnbj0l",
    "expiry": 60000000000,
    "objectName": "testObject"
  },
  "duration": 345,
  "error": "InternalError: We encountered an internal error, please try again.\n\tstatus code: 500, request id: 171489B93CFFE2FA, host id: ",
  "function": "/mint/build/versioning/put.go:83: main.testPutObject",
  "message": "PUT expected to succeed but got InternalError: We encountered an internal error, please try again.\n\tstatus code: 500, request id: 171489B93CFFE2FA, host id: ",
  "name": "versioning",
  "status": "FAIL"
}

Executed 9 out of 14 tests successfully.
```
<a name="29349102-14ba-496c-8200-cc8a31410678"></a>
### 15685-2c4b8ef/mint-compress-encrypt-dist-erasure.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-k8s6:30237
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0
RUN_ON_FAIL:          0

To get logs, run 'docker cp :/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 1 seconds
(2/14) Running aws-sdk-java tests ... done in 2 seconds
(3/14) Running aws-sdk-php tests ... done in 1 minutes and 31 seconds
(4/14) Running aws-sdk-ruby tests ... done in 5 seconds
(5/14) Running awscli tests ... done in 59 seconds
(6/14) Running healthcheck tests ... done in 0 seconds
(7/14) Running mc tests ... done in 28 seconds
(8/14) Running minio-go tests ... FAILED in 23 seconds
{
  "args": {
    "bucketName": "minio-go-test-lbz2aioesyl4x9bi",
    "objectName": "161pvsbgszxthbujby2bpokqkdt39b"
  },
  "duration": 5262,
  "error": "We encountered an internal error, please try again.",
  "function": "GetObject()",
  "message": "PutObject failed",
  "name": "minio-go: testPutObjectWithVersioning",
  "status": "FAIL"
}
(8/14) Running minio-java tests ... FAILED in 2 seconds
{
  "name": "minio-java",
  "function": "setBucketVersioning()",
  "duration": 188,
  "status": "FAIL",
  "error": "error occurred\nErrorResponse(code = InvalidBucketState, message = A replication configuration is present on this bucket, bucket wide versioning cannot be suspended., bucketName = minio-java-test-276li1e, objectName = null, resource = /minio-java-test-276li1e, requestId = 171489B6B8599245, hostId = bc3f7107-550b-4444-baed-84c7599350b8)\nrequest={method=PUT, url=http://minio-k8s6:30237/minio-java-test-276li1e?versioning=, headers=Host: minio-k8s6:30237\nAccept-Encoding: identity\nUser-Agent: MinIO (Linux; amd64) minio-java/8.4.3\nContent-MD5: orZUUp7E9srl53Od8p1glA==\nx-amz-content-sha256: d7f1d7ed35d188f6a89d394cc2055f5910c14a3abccf41e808e7f7327d21e129\nx-amz-date: 20220913T214620Z\nAuthorization: AWS4-HMAC-SHA256 Credential=*REDACTED*/20220913/us-east-1/s3/aws4_request, SignedHeaders=content-md5;host;x-amz-content-sha256;x-amz-date, Signature=*REDACTED*\n}\nresponse={code=400, headers=Accept-Ranges: bytes\nContent-Length: 387\nContent-Security-Policy: block-all-mixed-content\nContent-Type: application/xml\nServer: MinIO\nStrict-Transport-Security: max-age=31536000; includeSubDomains\nVary: Origin\nVary: Accept-Encoding\nX-Amz-Request-Id: 171489B6B8599245\nX-Content-Type-Options: nosniff\nX-Xss-Protection: 1; mode=block\nDate: Tue, 13 Sep 2022 21:46:20 GMT\n}\n >>> [io.minio.S3Base$1.onResponse(S3Base.java:690), okhttp3.internal.connection.RealCall$AsyncCall.run(RealCall.kt:519), java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128), java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628), java.base/java.lang.Thread.run(Thread.java:829)]"
}
(8/14) Running minio-js tests ... FAILED in 51 seconds
{
  "name": "minio-js",
  "function": "Suspend versioning  on a bucket",
  "duration": 4,
  "status": "FAIL",
  "error": "S3Error: A replication configuration is present on this bucket, bucket wide versioning cannot be suspended. at Object.parseError (node_modules/minio/dist/main/xml-parsers.js:71:11) at /mint/run/core/minio-js/node_modules/minio/dist/main/transformers.js:166:22 at DestroyableTransform._flush (node_modules/minio/dist/main/transformers.js:90:10) at DestroyableTransform.prefinish (node_modules/readable-stream/lib/_stream_transform.js:129:10) at prefinish (node_modules/readable-stream/lib/_stream_writable.js:611:14) at finishMaybe (node_modules/readable-stream/lib/_stream_writable.js:620:5) at endWritable (node_modules/readable-stream/lib/_stream_writable.js:643:3) at DestroyableTransform.Writable.end (node_modules/readable-stream/lib/_stream_writable.js:571:22) at IncomingMessage.onend (internal/streams/readable.js:670:10) at endReadableNT (internal/streams/readable.js:1333:12) at processTicksAndRejections (internal/process/task_queues.js:82:21)"
}
(8/14) Running minio-py tests ... FAILED in 18 seconds
{
  "name": "minio-py:test_stat_object_version",
  "status": "FAIL",
  "args": {
    "bucket_name": "minio-py-test-5656d0ba-6a07-49c2-8b58-df3f27153139",
    "object_name": "25c56344-6e8d-4698-84f2-529b71c5d5f6",
    "length": 1048576,
    "data": "LimitedRandomReader(1 * MB)"
  },
  "message": "HTTPConnectionPool(host='minio-k8s6', port=30237): Max retries exceeded with url: /minio-py-test-5656d0ba-6a07-49c2-8b58-df3f27153139/25c56344-6e8d-4698-84f2-529b71c5d5f6 (Caused by ResponseError('too many 500 error responses'))",
  "error": "Traceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 126, in _call_test\n    func(log_entry, *args, **kwargs)\n  File \"/mint/run/core/minio-py/tests.py\", line 825, in test_stat_object_version\n    _test_stat_object(log_entry, sse, version_check=True)\n  File \"/mint/run/core/minio-py/tests.py\", line 777, in _test_stat_object\n    result = _CLIENT.put_object(\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 1755, in put_object\n    raise exc\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 1709, in put_object\n    return self._put_object(\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 1567, in _put_object\n    response = self._execute(\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 403, in _execute\n    return self._url_open(\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 266, in _url_open\n    response = self._http.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/poolmanager.py\", line 376, in urlopen\n    response = conn.urlopen(method, u.request_uri, **kw)\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 878, in urlopen\n    return self.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 878, in urlopen\n    return self.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 878, in urlopen\n    return self.urlopen(\n  [Previous line repeated 2 more times]\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 868, in urlopen\n    retries = retries.increment(method, url, response=response, _pool=self)\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/util/retry.py\", line 592, in increment\n    raise MaxRetryError(_pool, url, error or ResponseError(cause))\nurllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='minio-k8s6', port=30237): Max retries exceeded with url: /minio-py-test-5656d0ba-6a07-49c2-8b58-df3f27153139/25c56344-6e8d-4698-84f2-529b71c5d5f6 (Caused by ResponseError('too many 500 error responses'))\n",
  "duration": 6532
}
(8/14) Running s3cmd tests ... done in 26 seconds
(9/14) Running s3select tests ... done in 4 seconds
(10/14) Running versioning tests ... FAILED in 1 seconds
{
  "alert": "",
  "args": {
    "bucketName": "versioning-test-4mzo0pepitmkx9",
    "expiry": 60000000000,
    "objectName": "testObject"
  },
  "duration": 491,
  "error": "InternalError: We encountered an internal error, please try again.\n\tstatus code: 500, request id: 171489CDFB5AD314, host id: ",
  "function": "/mint/build/versioning/put.go:83: main.testPutObject",
  "message": "PUT expected to succeed but got InternalError: We encountered an internal error, please try again.\n\tstatus code: 500, request id: 171489CDFB5AD314, host id: ",
  "name": "versioning",
  "status": "FAIL"
}

Executed 9 out of 14 tests successfully.
```
<a name="82976513-a24c-4b2a-a6c9-a546ace72384"></a>
### 15685-2c4b8ef/mint-large-bucket.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-k8s4:30776
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0
RUN_ON_FAIL:          0

To get logs, run 'docker cp :/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 8 seconds
(2/14) Running aws-sdk-java tests ... done in 1 seconds
(3/14) Running aws-sdk-php tests ... done in 51 seconds
(4/14) Running aws-sdk-ruby tests ... done in 4 seconds
(5/14) Running awscli tests ... done in 55 seconds
(6/14) Running healthcheck tests ... done in 0 seconds
(7/14) Running mc tests ... done in 27 seconds
(8/14) Running minio-go tests ... FAILED in 22 seconds
{
  "args": {
    "bucketName": "minio-go-test-oayz9srlcicd0ju6",
    "objectName": "u4rgfxhfjbmtoo1n9hrm0h2tfu3awp"
  },
  "duration": 5744,
  "error": "We encountered an internal error, please try again.",
  "function": "GetObject()",
  "message": "PutObject failed",
  "name": "minio-go: testPutObjectWithVersioning",
  "status": "FAIL"
}
(8/14) Running minio-java tests ... FAILED in 3 seconds
{
  "name": "minio-java",
  "function": "setBucketVersioning()",
  "duration": 193,
  "status": "FAIL",
  "error": "error occurred\nErrorResponse(code = InvalidBucketState, message = A replication configuration is present on this bucket, bucket wide versioning cannot be suspended., bucketName = minio-java-test-13soec6, objectName = null, resource = /minio-java-test-13soec6, requestId = 171489AA5062BCF8, hostId = e69e4d47-4d31-4932-838a-8d12a69f9d4e)\nrequest={method=PUT, url=http://minio-k8s4:30776/minio-java-test-13soec6?versioning=, headers=Host: minio-k8s4:30776\nAccept-Encoding: identity\nUser-Agent: MinIO (Linux; amd64) minio-java/8.4.3\nContent-MD5: orZUUp7E9srl53Od8p1glA==\nx-amz-content-sha256: d7f1d7ed35d188f6a89d394cc2055f5910c14a3abccf41e808e7f7327d21e129\nx-amz-date: 20220913T214526Z\nAuthorization: AWS4-HMAC-SHA256 Credential=*REDACTED*/20220913/us-east-1/s3/aws4_request, SignedHeaders=content-md5;host;x-amz-content-sha256;x-amz-date, Signature=*REDACTED*\n}\nresponse={code=400, headers=Accept-Ranges: bytes\nContent-Length: 387\nContent-Security-Policy: block-all-mixed-content\nContent-Type: application/xml\nServer: MinIO\nStrict-Transport-Security: max-age=31536000; includeSubDomains\nVary: Origin\nVary: Accept-Encoding\nX-Amz-Request-Id: 171489AA5062BCF8\nX-Content-Type-Options: nosniff\nX-Xss-Protection: 1; mode=block\nDate: Tue, 13 Sep 2022 21:45:26 GMT\n}\n >>> [io.minio.S3Base$1.onResponse(S3Base.java:690), okhttp3.internal.connection.RealCall$AsyncCall.run(RealCall.kt:519), java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128), java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628), java.base/java.lang.Thread.run(Thread.java:829)]"
}
(8/14) Running minio-js tests ... FAILED in 51 seconds
{
  "name": "minio-js",
  "function": "Suspend versioning  on a bucket",
  "duration": 5,
  "status": "FAIL",
  "error": "S3Error: A replication configuration is present on this bucket, bucket wide versioning cannot be suspended. at Object.parseError (node_modules/minio/dist/main/xml-parsers.js:71:11) at /mint/run/core/minio-js/node_modules/minio/dist/main/transformers.js:166:22 at DestroyableTransform._flush (node_modules/minio/dist/main/transformers.js:90:10) at DestroyableTransform.prefinish (node_modules/readable-stream/lib/_stream_transform.js:129:10) at prefinish (node_modules/readable-stream/lib/_stream_writable.js:611:14) at finishMaybe (node_modules/readable-stream/lib/_stream_writable.js:620:5) at endWritable (node_modules/readable-stream/lib/_stream_writable.js:643:3) at DestroyableTransform.Writable.end (node_modules/readable-stream/lib/_stream_writable.js:571:22) at IncomingMessage.onend (internal/streams/readable.js:670:10) at endReadableNT (internal/streams/readable.js:1333:12) at processTicksAndRejections (internal/process/task_queues.js:82:21)"
}
(8/14) Running minio-py tests ... FAILED in 20 seconds
{
  "name": "minio-py:test_stat_object_version",
  "status": "FAIL",
  "args": {
    "bucket_name": "minio-py-test-ac0ac880-4067-40b3-9006-bef9f799dc35",
    "object_name": "e976f2f3-5874-42b5-9f4e-e43142052bc8",
    "length": 1048576,
    "data": "LimitedRandomReader(1 * MB)"
  },
  "message": "HTTPConnectionPool(host='minio-k8s4', port=30776): Max retries exceeded with url: /minio-py-test-ac0ac880-4067-40b3-9006-bef9f799dc35/e976f2f3-5874-42b5-9f4e-e43142052bc8 (Caused by ResponseError('too many 500 error responses'))",
  "error": "Traceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 126, in _call_test\n    func(log_entry, *args, **kwargs)\n  File \"/mint/run/core/minio-py/tests.py\", line 825, in test_stat_object_version\n    _test_stat_object(log_entry, sse, version_check=True)\n  File \"/mint/run/core/minio-py/tests.py\", line 777, in _test_stat_object\n    result = _CLIENT.put_object(\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 1755, in put_object\n    raise exc\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 1709, in put_object\n    return self._put_object(\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 1567, in _put_object\n    response = self._execute(\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 403, in _execute\n    return self._url_open(\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 266, in _url_open\n    response = self._http.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/poolmanager.py\", line 376, in urlopen\n    response = conn.urlopen(method, u.request_uri, **kw)\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 878, in urlopen\n    return self.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 878, in urlopen\n    return self.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 878, in urlopen\n    return self.urlopen(\n  [Previous line repeated 2 more times]\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 868, in urlopen\n    retries = retries.increment(method, url, response=response, _pool=self)\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/util/retry.py\", line 592, in increment\n    raise MaxRetryError(_pool, url, error or ResponseError(cause))\nurllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='minio-k8s4', port=30776): Max retries exceeded with url: /minio-py-test-ac0ac880-4067-40b3-9006-bef9f799dc35/e976f2f3-5874-42b5-9f4e-e43142052bc8 (Caused by ResponseError('too many 500 error responses'))\n",
  "duration": 6522
}
(8/14) Running s3cmd tests ... done in 25 seconds
(9/14) Running s3select tests ... done in 4 seconds
(10/14) Running versioning tests ... FAILED in 1 seconds
{
  "alert": "",
  "args": {
    "bucketName": "versioning-test-ozcf1ojpt4l6vh",
    "expiry": 60000000000,
    "objectName": "testObject"
  },
  "duration": 408,
  "error": "InternalError: We encountered an internal error, please try again.\n\tstatus code: 500, request id: 171489C1E98B00D9, host id: ",
  "function": "/mint/build/versioning/put.go:83: main.testPutObject",
  "message": "PUT expected to succeed but got InternalError: We encountered an internal error, please try again.\n\tstatus code: 500, request id: 171489C1E98B00D9, host id: ",
  "name": "versioning",
  "status": "FAIL"
}

Executed 9 out of 14 tests successfully.
```
Deleting image on docker hub
Deleting image locally

--- a/cmd/bucket-object-lock.go
+++ b/cmd/bucket-object-lock.go
@@ -19,6 +19,7 @@ package cmd
 
 import (
 	"context"
+	"errors"
 	"math"
 	"net/http"
 
@@ -46,11 +47,14 @@ func (sys *BucketObjectLockSys) Get(bucketName string) (r objectlock.Retention, 
 
 	config, _, err := globalBucketMetadataSys.GetObjectLockConfig(bucketName)
 	if err != nil {
-		if _, ok := err.(BucketObjectLockConfigNotFound); ok {
+		if errors.Is(err, BucketObjectLockConfigNotFound{Bucket: bucketName}) {
 			return r, nil
 		}
+		if errors.Is(err, errInvalidArgument) {
+			return r, err
+		}
+		logger.CriticalIf(context.Background(), err)
 		return r, err
-
 	}
 	return config.ToRetention(), nil
 }

--- a/cmd/bucket-replication.go
+++ b/cmd/bucket-replication.go
@@ -81,6 +81,12 @@ func getReplicationConfig(ctx context.Context, bucketName string) (rc *replicati
 	}
 
 	rCfg, _, err := globalBucketMetadataSys.GetReplicationConfig(ctx, bucketName)
+	if err != nil {
+		if errors.Is(err, BucketReplicationConfigNotFound{Bucket: bucketName}) || errors.Is(err, errInvalidArgument) {
+			return rCfg, err
+		}
+		logger.CriticalIf(ctx, err)
+	}
 	return rCfg, err
 }
 
@@ -2325,7 +2331,7 @@ func QueueReplicationHeal(ctx context.Context, bucket string, oi ObjectInfo) {
 	if oi.VersionID == "" || oi.ModTime.IsZero() {
 		return
 	}
-	rcfg, _, _ := globalBucketMetadataSys.GetReplicationConfig(ctx, bucket)
+	rcfg, _ := getReplicationConfig(ctx, bucket)
 	tgts, _ := globalBucketTargetSys.ListBucketTargets(ctx, bucket)
 	queueReplicationHeal(ctx, bucket, oi, replicationConfig{
 		Config:  rcfg,

